compose/service.py

```diff
@@ -678,10 +678,21 @@ def labels(self, one_off=False):
         ]
 
     def can_be_scaled(self):
-        for port in self.options.get('ports', []):
-            if ':' in str(port):
+        # TODO: this function should ideally move to docker.utils.ports
+        # get rid of it once its accepted into docker-py
+        def has_external_port(external_binding):
+            if external_binding is None:
```

Comment:
We don't need to check this at all IMO because [docker-py's `split_port`](https://github.com/docker/docker-py/blob/master/docker/utils/ports/ports.py#L55) never returns a list with `None` elements as the 2nd return value.